### PR TITLE
chore(github-action): Refine GitHub Actions criteria and add Build & Push Actions 

### PR DESCRIPTION
**Pull request by niladrih**
Signed-off-by: Niladri Halder <niladri.halder@mayadata.io>

The build.yml and pull_request.yml workflows will ignore pushes made to `docs/` and `changelogs/` directories, all `.md` files, BUILDMETA, LICENSE, MAINTAINERS and VERSION files.

Added criteria to ignore build workflows on release. 

Adds Build & Push via Actions instead of makefile.

### CODE REVIEW

**Comment by niladrih** [comment]
Related PRs:
https://github.com/openebs/node-disk-manager/pull/562
https://github.com/openebs/cstor-operators/pull/271
https://github.com/openebs/dynamic-localpv-provisioner/pull/41
https://github.com/openebs/zfs-localpv/pull/306
https://github.com/openebs/lvm-localpv/pull/42
https://github.com/openebs/jiva-operator/pull/66
https://github.com/openebs/cstor-csi/pull/151
https://github.com/openebs/jiva/pull/350
https://github.com/openebs/libcstor/pull/85
https://github.com/openebs/velero-plugin/pull/151
https://github.com/openebs/m-exporter/pull/14

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -14,9 +14,9 @@
 name: release
 
 on:
-  create:
-    tags:
-      - 'v*'
+  release:
+    types:
+      - 'created'
```
Comment (akhilerm) [inline .github/workflows/release.yml:20]: Can you also add the tags here.

**Review by akhilerm — COMMENTED** [review]
Tags are missing from release workflow

**Review by akhilerm — APPROVED** [review]
LGTM

**Review by shubham14bajpai — APPROVED** [review]
LGTM

**Review by prateekpandey14 — APPROVED** [review]
lgtm, quay.io credential needs to be configured.

**Pull request merged**